collector/exporter/otelarrowexporter/internal/arrow/stream.go

```diff
@@ -287,7 +238,12 @@ func (s *Stream) run(bgctx context.Context, streamClient StreamClientFunc, grpcO
 // write repeatedly places this stream into the next-available queue, then
 // performs a blocking send().  This returns when the data is in the write buffer,
 // the caller waiting on its error channel.
-func (s *Stream) write(ctx context.Context) error {
+func (s *Stream) write(ctx context.Context) (retErr error) {
+	// always close send()
+	defer func() {
+		s.client.CloseSend()
+	}()
```

Comment:
This is as much about cleaning up tests as it is about being consistent when the stream is shutting down. Always closing the send channel appears to be working and it's simpler.